### PR TITLE
Update RDS aurora app implementation to execute mysql Query from pod

### PR DESCRIPTION
**Pull request by akankshakumari393**
## Change Overview

Currently, we had assumed that the test RDS instance would be accessible publicly, and we utilized mysql drivers to execute queries in the database. However, this approach would not have been viable if we were dealing with private instances. In order to address this issue, this PR alters the implementation of the RDS application, allowing for queries to be executed using mysql binary from a deployment resource instead.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/app/rds_aurora_mysql.go [hunk 1]
```diff
@@ -201,156 +212,77 @@ func (a *RDSAuroraMySQLDB) IsReady(context.Context) (bool, error) {
...
-	}()
+func (a *RDSAuroraMySQLDB) Ping(ctx context.Context) error {
+	log.Print("Pinging rds aurora database", field.M{"app": a.name})
+	isReadyQuery := fmt.Sprintf(mysqlConnectionString+"'SELECT 1;'", a.host, a.username, a.password, a.dbName)
 
-	pingQuery := "select 1"
-	_, err = db.Query(pingQuery)
-	return err
-}
+	pingCommand := []string{"sh", "-c", isReadyQuery}
```
Comment (viveksinghggits) [inline pkg/app/rds_aurora_mysql.go:219]: should we change either of these variables to have `isReady` or `ping`?

Comment (viveksinghggits) [reply pkg/app/rds_aurora_mysql.go:219]: If applicable for other PRs as well.

File: pkg/app/rds_aurora_mysql.go [hunk 2]
```diff
@@ -201,156 +212,77 @@ func (a *RDSAuroraMySQLDB) IsReady(context.Context) (bool, error) {
...
+func (a *RDSAuroraMySQLDB) Count(ctx context.Context) (int, error) {
+	log.Print("Counting entries from database", field.M{"app": a.name})
+	countQuery := fmt.Sprintf(mysqlConnectionString+
+		"\"SELECT COUNT(*) FROM pets;\"", a.host, a.username, a.password, a.dbName)
+
+	countCommand := []string{"sh", "-c", countQuery}
+	stdout, stderr, err := a.execCommand(ctx, countCommand)
 	if err != nil {
-		return 0, err
+		return 0, errors.Wrapf(err, "Error while counting data into table: %s, app: %s", stderr, a.name)
```
Comment (viveksinghggits) [inline pkg/app/rds_aurora_mysql.go:252]:
```suggestion
		return 0, errors.Wrapf(err, "Error while counting data of table: %s, app: %s", stderr, a.name)
```

File: pkg/app/rds_aurora_mysql.go [hunk 3]
```diff
@@ -201,156 +212,77 @@ func (a *RDSAuroraMySQLDB) IsReady(context.Context) (bool, error) {
...
 
-	rows, err := db.Query("select * from pets;")
+	rowsReturned, err := strconv.Atoi(stdout)
 	if err != nil {
-		return 0, errors.Wrap(err, "Error preparing count query")
-	}
-	count := 0
-	for rows.Next() {
-		count++
+		return 0, errors.Wrapf(err, "Error while converting response of count query: %s, app: %s", stderr, a.name)
```
Comment (viveksinghggits) [inline pkg/app/rds_aurora_mysql.go:257]:
```suggestion
		return 0, errors.Wrapf(err, "Error while converting response of count query to int: %s, app: %s", stderr, a.name)
```

File: pkg/app/rds_aurora_mysql.go [hunk 4]
```diff
@@ -201,156 +212,77 @@ func (a *RDSAuroraMySQLDB) IsReady(context.Context) (bool, error) {
...
 	}
 
-	return count, nil
+	log.Info().Print("Counting rows in test db.", field.M{"app": a.name, "count": rowsReturned})
```
Comment (viveksinghggits) [inline pkg/app/rds_aurora_mysql.go:260]:
```suggestion
	log.Info().Print("Counting rows in test db.", field.M{"app": a.name, "count": rowsReturned})
```

should this be, `Number of rows in test DB`? Or are we still counting the rows.

File: pkg/app/rds_aurora_mysql.go [hunk 5]
```diff
@@ -201,156 +212,77 @@ func (a *RDSAuroraMySQLDB) IsReady(context.Context) (bool, error) {
...
-	}
-
-	_, err = tx.Stmt(query).Exec()
-	if err != nil {
-		return errors.Wrap(err, "Error resetting Aurora database")
-	}
-
-	if err = tx.Commit(); err != nil {
-		return errors.Wrap(err, "Error committing DB reset transaction")
+		return errors.Wrapf(err, "Error while deleting data into table: %s, app: %s", stderr, a.name)
```
Comment (viveksinghggits) [inline pkg/app/rds_aurora_mysql.go:271]:
```suggestion
		return errors.Wrapf(err, "Error while deleting data from table: %s, app: %s", stderr, a.name)
```

File: pkg/app/rds_aurora_mysql.go [hunk 6]
```diff
@@ -201,156 +212,77 @@ func (a *RDSAuroraMySQLDB) IsReady(context.Context) (bool, error) {
...
 	}
 
+	log.Info().Print("Database reset successful!", field.M{"app": a.name})
```
Comment (viveksinghggits) [inline pkg/app/rds_aurora_mysql.go:274]:
```suggestion
	log.Info().Print("Database reset was successful!", field.M{"app": a.name})
```

File: pkg/app/rds_aurora_mysql.go [hunk 7]
```diff
@@ -201,156 +212,77 @@ func (a *RDSAuroraMySQLDB) IsReady(context.Context) (bool, error) {
...
 	return nil
 }
 
-func (a *RDSAuroraMySQLDB) Initialize(context.Context) error {
-	db, err := a.openDBConnection()
+func (a *RDSAuroraMySQLDB) Initialize(ctx context.Context) error {
+	// Create table.
```
Comment (viveksinghggits) [inline pkg/app/rds_aurora_mysql.go:279]:
```suggestion
```

**Review by PrasadG193 — APPROVED** [review]
LGTM

**Pull request merged**